ethereum/solidity/contracts/QuantumGravityBridge.sol

```diff
@@ -16,29 +20,33 @@ struct Signature {
     bytes32 s;
 }
 
-struct MessageTuple {
-    uint256 height;
-    bytes8 namespaceID;
-    bytes32 messageCommitment;
-}
-
 /// @title Quantum Gravity Bridge: Celestia -> Ethereum, Data Availability relay.
-contract QuantumGravityBridge is OwnableUpgradeableWithExpiry {
-    // Don't change the order of state for working upgrades.
-    // AND BE AWARE OF INHERITANCE VARIABLES!
-    // Inherited contracts contain storage slots and must be accounted for in any upgrades
-    // always test an exact upgrade on testnet and localhost before mainnet upgrades.
+/// @dev The relay relies on a set of signers to attest to some event on
+/// Celestia. These signers are the Celestia validator set, who sign over every
+/// Celestia block. Keeping track of the Celestia validator set is accomplished
+/// by updating this contract's view of the validator set with
+/// `updateValidatorSet()`. At least 2/3 of the voting power of the current
+/// view of the validator set must sign off on new relayed events, submitted
+/// with `submitDataRootTupleRoot()`. Each event is a batch of `DataRootTuple`s
+/// (see ./DataRootTuple.sol), with each tuple representing a single data root
+/// in a Celestia block header. Relayed tuples are in the same order as the
+/// block headers.
+contract QuantumGravityBridge is IDAOracle, OwnableUpgradeableWithExpiry {
+    // Don't change the order of state for working upgrades AND BE AWARE OF
+    // INHERITANCE VARIABLES! Inherited contracts contain storage slots and must
+    // be accounted for in any upgrades. Always test an exact upgrade on testnet
+    // and localhost before mainnet upgrades.
 
     ///////////////
     // Constants //
     ///////////////
 
-    // bytes32 encoding of the string "checkpoint"
+    /// @dev bytes32 encoding of the string "checkpoint"
     bytes32 constant VALIDATOR_SET_HASH_DOMAIN_SEPARATOR =
         0x636865636b706f696e7400000000000000000000000000000000000000000000;
 
-    // bytes32 encoding of the string "transactionBatch"
-    bytes32 constant MESSAGE_TUPLE_ROOT_DOMAIN_SEPARATOR =
+    /// @dev bytes32 encoding of the string "transactionBatch"
```

Comment:
What is this `@dev` annotation?

Reply:
It's [Natspec](https://docs.soliditylang.org/en/v0.8.4/natspec-format.html), a documentation format for Solidity. It's terrible.

Reply:
Oh I see. I think I watched Franzi giving a presentation about making it more usable.